### ISSUE TITLE
Add a way to set BASE_URL for frontend

### ISSUE THREAD

**Issue by serbaut** [coreroller/coreroller#58]
This would make it possible to run the coreroller web and api under e.g. /test/coreroller/ instead of /.

**Comment by tegioz** [comment]
Hi @serbaut 

Could you explain a bit more the use case for this?

**Comment by serbaut** [comment]
We mount coreroller under /something/coreroller/ using url rewrites. We do the same for /something/kibana/ and /something/grafana/. Grafana is a pain since it has to recompile the frontend if the path changes which takes minutes but i can see that you have a BASE_URL variable that works fine, I just need to be able to set it somehow.

**Comment by serbaut** [comment]
This is how our kubernetes ingress for coreroller looks:

```
apiVersion: extensions/v1beta1
kind: Ingress
metadata:
  name: rollerd
  annotations:
    ingress.kubernetes.io/rewrite-target: /
spec:
  rules:
  - http:
      paths:
      - path: /comething/coreroller/
        backend:
          serviceName: rollerd
          servicePort: 80
```
and the resulting nginx config:

        rewrite /something/coreroller/(.*) /$1 break;
        proxy_pass http://default-rollerd-80;

This works with  `BASE_URL=/something/coreroller/api` in API.js.

**Comment by tegioz** [comment]
Thanks for your detailed explanation.

CoreRoller frontend is a single page application that contains that BASE_URL hardcoded in the JS code. One way of changing it would be to replace it *before* the frontend is built adding a `sed` line to the sample `rollerd` dockerfile (around lines below would be a good place).

https://github.com/coreroller/coreroller/blob/5ec1a68309b9c982335dd2fc5a1ebf72f77b9c40/backend/docker/rollerd/Dockerfile#L12-L15

Please note that there is [one place](https://github.com/coreroller/coreroller/blob/b3247330fa666f6590705753c2168c5d51f7f007/frontend/src/js/api/API.js#L14) where BASE_URL wasn't being used (it's harcoding `/api` incorrectly). This will be addressed soon.

On the kubernetes side, do you think using [name based virtual hosting](https://kubernetes.io/docs/concepts/services-networking/ingress/#name-based-virtual-hosting) would help? Maybe it's not an option in your case but it may save you from having to manipulate the paths to route the traffic to the right backend. It's very handy too when handling multiple services with the same ingress resource.

**Comment by serbaut** [comment]
@tegioz Thats more or less what I have done as a workaround but I would like this to be configurable in runtime, not compile time.

Name based virtual hosting would probably work but not for us.

**Comment by tegioz** [comment]
BASE_URL is now used in all frontend api calls (fixed in #62). I'll close this one for now @serbaut. If a nice solution to make it configurable at runtime comes up we can reopen it. The workaround you are using seems reasonable though.
